Improve instructions for getting started with Docker.

bfinney-thoughtworks opened the pull request:
The instructions failed for me because they didn't include the “install and start Docker” step.

This merge request makes it explicit how that's meant to happen.

Commit:
Refine formatting of instruction steps.

Commit:
Explicitly state the “install and start Docker” step.

pameck commented on README.md:
```diff
@@ -18,21 +18,40 @@ To find out more about the Rabble Rouser project, check out our [documentation r
         git clone https://github.com/rabblerouser/core.git
 
-3. Start a Docker container to develop in (this also starts containers for dependent services):
+3. Install Docker and start its application.
```
This is already described in Step 1 in the readme, including the links. We could have a step to verify that the docker daemon is running? 

3. Verify Docker is running

`$ docker info`

bfinney-thoughtworks replied:
Right, I did that step but then Docker wasn't running, and later instructions failed. By the time I realised the problem, I forgot that I'd installed Docker :-)

Updated this merge request to follow on from the existing “install Docker” instruction.

pameck reviewed (commented):
Hi @bfinney-thoughtworks I think this PR helps clarifying the setup process, thanks. I left a comment in the code about a step I think is duplicated.

Commit:
Make a “start the Docker application” step.

The “install Docker” instruction was already present, so remove the
extraneous instruction.
